File: src/main/java/com/kelsos/mbrc/commands/RestartConnectionCommand.java
package com.kelsos.mbrc.commands;

import com.google.inject.Inject;
import com.kelsos.mbrc.enums.SocketAction;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.services.SocketService;

public class RestartConnectionCommand implements ICommand
{
	@Inject
	SocketService socket;
	@Override
	public void execute(IEvent e)
	{
		socket.SocketManager(SocketAction.RETRY_COUNTER_RESET);
		socket.SocketManager(SocketAction.SOCKET_RESET);
	}
}

File: app/src/main/java/com/kelsos/mbrc/commands/RestartConnectionCommand.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.annotations.SocketAction.RESET
import com.kelsos.mbrc.networking.client.SocketService
import com.kelsos.mbrc.networking.protocol.ProtocolAction
import com.kelsos.mbrc.networking.protocol.ProtocolMessage

class RestartConnectionCommand(
  private val socket: SocketService,
) : ProtocolAction {
  override fun execute(message: ProtocolMessage) {
    socket.socketManager(RESET)
  }
}
